src/slick.grid.ts

```diff
@@ -2448,10 +2448,10 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
       for (i = 0; i < cssRules.length; i++) {
         const selector = cssRules[i].selectorText;
         if (matches = /\.l\d+/.exec(selector)) {
-          columnIdx = parseInt(matches[0].substr(2, matches[0].length - 2), 10);
+          columnIdx = parseInt(matches[0].slice(2), 10);
           this.columnCssRulesL[columnIdx] = cssRules[i];
         } else if (matches = /\.r\d+/.exec(selector)) {
-          columnIdx = parseInt(matches[0].substr(2, matches[0].length - 2), 10);
+          columnIdx = parseInt(matches[0].slice(2), 10);
```

Comment:
why are you replacing `substr` with `slice`? I'm not sure if this is really safe to use and even if it is, this change should **not** be part of this PR since this is unrelated to your original issue. PR should always focus on 1 thing not multiple. Also please explain the reason to use slice before creating another PR

Reply:
string.prototype.substr is deprecated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr
string.prototype.slice is also supportet on all browsers including IE.
https://caniuse.com/?search=slice

I did not touch other deprecated properties like Event.wich, Window.pageYOffset and Window.pageXOffset, because of possible IE problems.

Reply:
well that's the thing about native JS, even if it's deprecated they will probably never remove it. Why? Because as I often heard from NodeJS core developers, they can't break the web, so a website that worked 20 years ago will still work 20 years from now... Same goes for `e.keyCode` (Event) and other deprecated code that is used across SlickGrid, we could replace it in the future but there's really no rush in doing so without a lot of testing